### ISSUE TITLE
White Dots / Screen Tearing

### ISSUE THREAD

**Issue by Jack-Follows** [RenderHeads/UnityPlugin-AVProVideo#1972]
Hi I am currently developing for Meta Quest 2.

I am running 8k videos in the app and I am experiencing white dots appearing randomly in both eyes occasionally. The right eye does have some screen tearing also.

Unity version: 2022.3.36f1
AVPro version: AVPro Video v3 - Core Mobile Edition (3.0.8)
Operating System version: SQ3A.220605.009.A1
Device Model: Meta Quest 2
Video Specs: 8k, 30fps, HEVC, 3.5gb

I have followed the set up for OES however I am still receiving this error.

<img width="960" alt="AVPro_Issue" src="https://github.com/user-attachments/assets/4a7dd13d-cdc6-421f-a848-d857a31ed578">

I have also applied the OES material to the 360 video sphere in the level to ensure I am matching this.

Any time a video is played I am having this issue. I have tried on multiple videos but just cannot seem to find a fix. I have looked at previous issues on this topic but their fixes don't seem to work. Am I missing something?

https://github.com/RenderHeads/UnityPlugin-AVProVideo/issues/1416

I had attempted to enable the Stereo Debug Tinting however this did not show any tints at all (I did test in a build, I am unsure if this option only works in editor?).

Thanks in advance!

**Comment by Jack-Follows** [comment]
Hi Chris,

Sure thing, I've conducted a recording however it doesn't seem anywhere near as bad on the recording compared to the actual experience.

Recording:

[cid:bbec30d9-1912-4dbd-bf4b-b4f86de19f85]AVProIssue.mp4<https://1drv.ms/v/c/55543f20152b1ffd/Edaw5Pn8zphHnLaqY40kv9EBVSyJ0JSr8Bfai-IMnAqeVA>

APK build tested on Meta Quest 2:

[https://res.cdn.office.net/assets/mail/file-icon/png/exe_16x16.png]GoldVRV101.apk<https://1drv.ms/u/c/55543f20152b1ffd/EVIOPnwCZelHr7pGcb8jVxEB5idTNcpmvbpZiqqEPrtbIw>

The app is designed to read files in from the movies folder on the headset and doesn't have any media embedded in the app.

The video file is too large to send over here, would you have any alternative routes to receive data?

I also have a zip file with the code included but again is too large to send over through Outlook/OneDrive.

All the above files are shared in confidence and under the understanding that these files will not be shared with any third-party companies or replicated otherwise. I'm sure you wouldn't but I just wanted to make sure I said it as there is code and videos which are both mine and my colleague's content.

If you need any further information, please do not hesitate to contact myself.

Thank you so much again for the speedy responses!

Look forward to your reply!

Kind regards,

Jack Follows.

________________________________

**Comment by Chris-RH** [comment]
Hi, 

Please could you resend those files to our email address: unitysupport@renderheads.com
A lot of devs use googledrive or dropbox for larger files.

Kind regards, 
Chris

**Comment by Chris-RH** [comment]
Hi,

Please could you resend those files to our email address: [unitysupport@renderheads.com](mailto:unitysupport@renderheads.com)
A lot of devs use googledrive or dropbox for larger files.

Kind regards,
Chris

**Comment by Jack-Follows** [comment]
Hi Chris, apologies as I’ve been on holiday, leave it with me and I’ll get these sent across to you today!

Kind regards,

Jack Follows.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Jack-Follows** [comment]
Hi Chris,

Just checking in to see if you received the code base and the video that I shared across google drive?

Kind regards,

Jack Follows.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Ste-RH** [comment]
We requested access to the files last week.  Access has still not been granted.

**Comment by Ste-RH** [comment]
Your video is 75 mbit/sec.  That is very high for a Quest2.  Have you tried something a lot lower to see if the issue still happens?  The lower the better for a test, try something like 10 mbit and if the problem goes away, increase it a bit - maybe up towards 40 mbit.

**Comment by Jack-Follows** [comment]
Brilliant, thanks for this info, will pass this onto my content creator so they can edit the videos accordingly. Once we have retested the videos, I’ll be back in contact to let you know how it goes!

Thanks again!

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Jack-Follows** [comment]
Hi Ste,

Just to follow on from your advice for the different spec videos, we tested, 10, 25, and 40 however the issue still persists.

The quality of the video was also massively reduced below 50 Mbps bitrate. This wouldn't be suitable for our needs as we need higher quality videos. Based on the YouTuber Hugh Hou (professional 360 video creator), he recommends for 8k videos to be played on Quest 2 at 100 Mbps bitrate as that's the maximum recommended (optimal) bitrate to run at.

Are there some settings either in C# or the editor that could be tweaked to accommodate 8k videos?

**Comment by Ste-RH** [comment]
There is no way you will get 8k @ 100mbit played back perfectly on a Quest2 when using Unity as the playback engine.  Unity adds a fair amount of overhead compared to using a standalone video player application.

Have you tried lowering the resolution to 4k - just to check the issue goes away.

**Comment by Chris-RH** [comment]
For more info, [here](https://creator.oculus.com/getting-started/media-production-specifications-for-delivery-to-meta-quest-2-headsets/?locale=en_GB) and [here](https://headjack.io/knowledge-base/best-video-resolution-for-oculus-quest-2/) are some guidelines for MAXIMUM encoding recommendations for the Quest 2.  Note, these are based on using built in players that will not have the added overhead that Unity has.

**Comment by Jack-Follows** [comment]
Hi both, thanks for the info! We are currently working on editing our videos to get the best quality at a lower res however, could video caching be our answer here in order to allow us to play higher quality (6k - 8k) videos at a higher bitrate without any issues?

If so what would be the best approach to ensure this is set up correctly to allow us the maximum capable stream from the headsets memory?

**Comment by Chris-RH** [comment]
Unfortunately caching won't help as the bottleneck is caused by decoding, not downloading.  The device cannot decode at the high resolution/bitrate on top of Unity's overheads.

**Comment by Jack-Follows** [comment]
Ahh okay, not to worry. Have you or the team tested videos in the Quest 2? If so would what would be the best settings in both Unity and the video output from Premiere Pro to achieve the best quality?

Also looking into the specs of the Quest 3, it seems there is a much higher processing speed in both CPU and GPU. Has any of the team tested with Quest 3 and if so would that be capable of 8K+ videos?

**Comment by Chris-RH** [comment]
Its really difficult to say.  We can't really say how your particular use case would push the hardware.  In theory, you should be able push close to the recommended specs, but you need to make sure that you leave a bit of wiggle room.
Its the type of thing that needs to be up to the individual developer.  To test various pieces of hardware against their own use case and requirements.

**Comment by Jack-Follows** [comment]
I know this issue has been closed however I had also tested on a Quest 3 and the videos within the app were playing worse. I did a bit of digging and found that using GLES 3.0 instead of Vulkan had rectified the issue and is now working extremely well! It is now playing videos at 100mbps+ with no issues on both Quest 2 and 3 as good as a stand-alone video player. Just thought I'd pop this comment in as it could help others with their issues!
